dependencies.yaml

```diff
@@ -428,9 +406,6 @@ dependencies:
       match: OS_CODENAME\ \?=\ bullseye
     - path: images/build/cross/variants.yaml
       match: "OS_CODENAME: 'bullseye'"
-      # We keep them around for backwards compatibility and release branches.
-    - path: images/build/go-runner/variants.yaml
-      match: "OS_CODENAME: 'bullseye'"
```

Comment:
does updating to go1.21 imply dropping bullseye? that is not obvious to me.

Reply:
that was for go1.20, in go1.21+ we are using bookworm

Reply:
is that *required* for go1.21, or just incidental? does that impact anything other than our build / test images? if it's just for CI / build, that's probably fine. if it's something we ship in the release, should we try to limit ourselves to one change at a time (changing go without changing os image as well)?